docs/DeveloperGuide.md

```diff
@@ -2,21 +2,133 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links 
+to the original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The following is our class diagram for our whole project.
 
+![Local Image](UML/CLiRental.png)
+
+### Sequence diagrams
+
+1. `Adding a customer`
+
+The following sequence diagram will explain the sequence of events after the user inputs an add-user command.
+
+![Local Image](UML/AddCustomerSequence.png)
+
+2. `Creating Car file and loading file`
+
+The following sequence diagram will explain the sequence of events for loading of the carData.txt which happens at the
+start of the program. The carData.txt will be created if it does not exist at the start of program and its data will be
+loaded if the file exist.
+
+![Local Image](UML/CarFileLoader.png)
+
+### Auto updating of car rental status feature
+
+### Implementation:
+
+Every car added to the car list will have a default rental status of **'Available'**.
+Once a valid transaction has been made and added to the transaction list, the particular car,
+identified by its **unique identifier (i.e. license plate number)**, will automatically be
+marked as **'Rented'**.
+
+Given below is a step-by-step execution of the implementation:
+
+Step 1: The user launches the application and enters the `add-car ...` command to add a new
+car to the list, specifying the model name, car license plate number and price of the car.
+
+> **Note**: If the command fails due to invalid parameters or format,
+the car will not be added to the list.
+
+The following sequence diagram illustrates a **valid** `add-car` operation:
+
+![](./images/RyanAddCarDiagram.png)
+
+Step 2: The user enters the `list-cars` command to verify that the car has been successfully added
+to the car list.
+
+-> show code output
+
+Example:
+
+![](./images/list-cars-output-before.png)
+
+
+Step 3: A customer decides to rent a car from the company. The user then uses the application to track
+and record the transaction details. The user executes the `add-tx ...` command to add
+a new transaction record, specifying details like the car license plate number.
+
+> **Note**: If the command fails due to invalid parameters or format,
+the transaction will not be added to the list and the car rental status remains as **'Available'**.
+
+The following sequence diagram illustrates a **valid** `add-tx` operation:
+
+![](./images/RyanAddTxDiagram.png)
+
+Step 4: After adding the transaction, the rental status of the selected car will now be updated to **'Rented'**.
+The user finally executes the `list-rented` command to list all rented out cars. The car that
+was just rented out should appear in the list, together with other rented out cars (if any).
+
+Example:
+
+![](./images/list-rented-output.png)
+
+Step 5: **Optionally**, the user can also execute 
```

Comment:
I like the explanation of this process in steps, it helps to make it more readable.